### PR TITLE
Update Datastore query docs

### LINKED ISSUE #1357: [Datastore] Query namespaced entities
Hi all,

I had a customer question about how-to query namespaced entities. I couldn't find an example snippet in the gem documentation for Datastore Queries.

After reviewing [issue 404](https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/404), I found the following solution. This should be called out somewhere in the documentation.

```ruby
# Datastore service client
datastore = Google::Cloud::Datastore.new

# Create a new entry with a namespace
new_entry = datastore.entity "kind", namespace: "test-namespace" do |entry|
  entry["name"]  = "YOURNAME"
end

# Save in Datastore
datastore.save new_entry

# Query a namespaced entry
query = Google::Cloud::Datastore::Query.new
query.kind "kind"
datastore.run query, namespace: "test-namespace"
```

### PR DESCRIPTION
**Pull request by quartzmo**
Add the namespace example to the guide and Query.

[closes #1357]

### CODE REVIEW

**Comment by blowmage** [comment]
Waiting to hear from @frankyn before merging.

File: google-cloud-datastore/lib/google/cloud/datastore.rb [hunk 1]
```diff
@@ -130,6 +130,21 @@ module Cloud
...
+    #
+    # ```ruby
+    # require "google/cloud/datastore"
+    #
+    # datastore = Google::Cloud::Datastore.new
+    #
+    # query = datastore.query("Task").
+    #   where("done", "=", false)
+    #
+    # tasks = datastore.run query, namespace: "ns~todo-project"
```
Comment (frankyn) [inline google-cloud-datastore/lib/google/cloud/datastore.rb]: s/ns~todo-project/example-ns/g, maybe?

File: google-cloud-datastore/lib/google/cloud/datastore/query.rb [hunk 2]
```diff
@@ -42,6 +42,15 @@ module Datastore
...
       #   tasks = datastore.run query
       #
+      # @example Run the query within a namespace with the `namespace` option:
+      #   require "google/cloud/datastore"
+      #
+      #   datastore = Google::Cloud::Datastore.new
+      #
+      #   query = datastore.query("Task").
+      #     where("done", "=", false)
+      #   tasks = datastore.run query, namespace: "ns~todo-project"
```
Comment (frankyn) [inline google-cloud-datastore/lib/google/cloud/datastore/query.rb]: s/ns~todo-project/example-ns/g, maybe?

File: google-cloud-datastore/support/doctest_helper.rb [hunk 3]
```diff
@@ -282,6 +282,12 @@ def mock_datastore
     end
   end
 
+  doctest.before("Google::Cloud::Datastore::Query@Run the query within a namespace with the `namespace` option:") do
+    mock_datastore do |mock|
+      mock.expect :run_query, run_query_res, ["my-todo-project", Google::Datastore::V1::PartitionId, nil, Hash]
```
Comment (frankyn) [inline google-cloud-datastore/support/doctest_helper.rb]: I'm not sure how this tests the namespace parameter. Could you give me insight?

Comment (quartzmo) [inline google-cloud-datastore/support/doctest_helper.rb]: This block sets up the mocking to support executing the example. It doesn't test the namespace parameter. The goal is to verify that the code example executes.

Comment (frankyn) [inline google-cloud-datastore/support/doctest_helper.rb]: Thank you!

**Pull request merged**